### DISCUSSION TITLE
Electron using WalletConnect web3modal works on development but not built version

### DISCUSSION THREAD

**Discussion by tiagorvmartins** [WalletConnect/walletconnect-monorepo#2380] [Developer Support] [answered]
Hello, I am trying to have a built version of my electron application to be able to connect to my MetaMask wallet using WalletConnect.
It's working when I run the electron app in development mode (which hosts an http local server), and so requests come form http.

However, when building the app (executable)  and launching, the protocol from which the requests come are from file:// if I recall correctly, and I believe that's the issue currently for getting "session rejected" on WalletConnect popup.
Went through a bit of debugging in the console devtools, and I notice that the error on EventManager while parsing the response have this:

code: -32000
message: "Session Rejected"

I also notice that the request done on network was this one:
wss://v.bridge.walletconnect.org/?env=browser&host=&protocol=wc&version=1

As you can see the value of the key host is empty. So I believe that's related with the issue.
Thank you!

**Comment by xzilja** [comment, marked as answer]
To the best of my knowledge a lot of wallets check protocol and only allow http:// via localhost otherwise, only `https://`. Might be the case where wallets insta reject the session due to `file://` protocol.

**Reply by tiagorvmartins** [reply]
Thank you for your fast feedback!
